src/file_integrity.rs

```diff
@@ -0,0 +1,46 @@
+/// TODO: how to retrieve the actual pure checksum (reference value)
+use serde::{Deserialize, Serialize};
+
+/// Fictional struct for the moment / prob use an external library
+#[derive(PartialEq)]
+pub struct Checksum {
```

Comment:
*: Most types will want `Clone`, `Debug`, `PartialEq`, and `PartialOrd`

`Hash` is kinda optional, but I tend to add it in case I use `HashSet` over `Vec` somewhere


```suggestion
#[derive(Clone, Debug, PartialEq, PartialOrd)]
```